src/services/users.service.js

```
@@ -0,0 +1,55 @@
+let users = [];
+let nextId = 1;
+
+const getAll = () => {
+  return users;
+};
+
+const getById = (id) => {
+  return users.find((user) => user.id === parseInt(id)) || null;
+};
+
+const add = (user) => {
+  const createdUser = Object.assign({}, user, { id: nextId++ });
+
+  users.push(createdUser);
+
+  return createdUser;
+};
+
+const remove = (id) => {
+  const existing = getById(id);
+
+  if (!existing) {
+    return;
+  }
+  users = users.filter((user) => user.id !== parseInt(id));
```

remove silently returns when the entity doesn't exist (no explicit return value). That's acceptable because controllers check existence before calling remove, but you could return a boolean to indicate success/failure which can make the service API clearer.